failed to start prometheus

with single node setup,
```
python paictl.py service start -p ./my-quick-start
```

generate following error:
```
2018-07-27 12:11:20,984 [ERROR] - paiLibrary.paiService.service_template_generate : failed to generate template file from bootstrap/prometheus/alert-deployment.yaml.template with dict {'machinelist': {'stcvl-055': {'webportal': 'true', 'ip': '10.190.151.29', 'zookeeper': 'true', 'pai-master': 'true', 'zkid': '1', 'pylon': 'true', 'hadoop-resource-manager': 'true', 'machinetype': 'GENERIC', 'hostname': 'stcvl-055', 'grafana': 'true', 'pai-worker': 'true', 'username': 'zhdo', 'hadoop-data-node': 'true', 'nodename': '10.190.151.29', 'restserver': 'true', 'prometheus': 'true', 'hadoop-node-manager': 'true', 'password': 'Pass_word', 'etcdid': 'etcdid1', 'k8s-role': 'master', 'jobhistory': 'true', 'hadoop-name-node': 'true', 'launcher': 'true', 'dashboard': 'true', 'hostip': '10.190.151.29', 'sshport': 22, 'node-exporter': 'true', 'machine-type': 'GENERIC'}}, 'machineinfo': {'GENERIC': {'mem': 1, 'os': 'ubuntu16.04', 'cpu': {'vcore': 1}}}, 'clusterinfo': {'data-path': '/datastorage', 'prometheusinfo': {'prometheus_url': 'http://10.190.151.29', 'node-exporter-port': 9100, 'alert-manager-hosts': [], 'node_exporter_port': 9100, 'prometheus-port': 9091, 'prometheus_port': 9091}, 'restserverinfo': {'jwt_secret': 'pai-secret', 'hdfs_uri': 'hdfs://10.190.151.29:9000', 'default_pai_admin_password': 'admin-password', 'default-pai-admin-username': 'admin', 'etcd_uri': 'http://10.190.151.29:4001', 'webhdfs_uri': 'http://10.190.151.29:50070', 'src_path': '../rest-server/', 'server_port': 9186, 'yarn_uri': 'http://10.190.151.29:8088', 'server-port': 9186, 'webservice_uri': 'http://10.190.151.29:9086', 'default-pai-admin-password': 'admin-password', 'default_pai_admin_username': 'admin', 'jwt-secret': 'pai-secret'}, 'hadoopinfo': {'configmapname': 'hadoop-configuration', 'custom-hadoop-binary-path': '/pathHadoop/hadoop-2.9.0.tar.gz', 'custom_hadoop_binary_path': '/pathHadoop/hadoop-2.9.0.tar.gz', 'hadoopversion': '2.9.0', 'hadoop-version': '10.190.151.29', 'hadoop_vip': '10.190.151.29'}, 'grafanainfo': {'grafana_url': 'http://10.190.151.29', 'grafana-port': 3000, 'grafana_port': 3000}, 'clusterid': 'pai-example', 'dataPath': '/datastorage', 'dockerregistryinfo': {'credential': '{}', 'docker-namespace': 'openpai', 'secret-name': 'pai-secret', 'base64code': 'e30=', 'docker_password': None, 'secretname': 'pai-secret', 'docker_username': None, 'prefix': 'docker.io/openpai/', 'docker_tag': 'latest', 'docker-tag': 'latest', 'docker-registry-domain': 'docker.io', 'docker_namespace': 'openpai', 'docker_registry_domain': 'docker.io'}, 'hadoopQueues': {'default': {'description': 'Default VC.', 'weight': 100.0}}, 'webportalinfo': {'prometheus_uri': 'http://10.190.151.29:9091', 'k8s_dashboard_uri': 'http://10.190.151.29:9090', 'rest_server_uri': 'http://10.190.151.29:9186', 'grafana_uri': 'http://10.190.151.29:3000', 'server_port': 9286, 'k8s_api_server_uri': 'http://10.190.151.29:8080', 'server-port': 9286, 'yarn_web_portal_uri': 'http://10.190.151.29:8088'}, 'docker-registry-info': {'credential': '{}', 'docker-namespace': 'openpai', 'secret-name': 'pai-secret', 'base64code': 'e30=', 'docker_password': None, 'secretname': 'pai-secret', 'docker_username': None, 'prefix': 'docker.io/openpai/', 'docker_tag': 'latest', 'docker-tag': 'latest', 'docker-registry-domain': 'docker.io', 'docker_namespace': 'openpai', 'docker_registry_domain': 'docker.io'}, 'frameworklauncher': {'frameworklauncher_port': 9086, 'frameworklauncher_vip': '10.190.151.29', 'frameworklauncher-port': 9086}, 'pyloninfo': {'prometheus_uri': 'http://10.190.151.29:9091', 'k8s_dashboard_uri': 'http://10.190.151.29:9090', 'rest_server_uri': 'http://10.190.151.29:9186', 'webhdfs_uri': 'http://10.190.151.29:50070', 'pai_web_portal_uri': 'http://10.190.151.29:9286', 'grafana_uri': 'http://10.190.151.29:3000', 'k8s_api_server_uri': 'http://10.190.151.29:8080', 'port': 80, 'yarn_web_portal_uri': 'http://10.190.151.29:8088'}, 'virtualClusters': {'default': {'capacity': 1, 'description': 'Default VC.'}}}}
Traceback (most recent call last):
  File "/pai/pai-management/paiLibrary/paiService/service_template_generate.py", line 82, in generate_template
    generated_template = template_handler.generate_from_template_dict(template_data, service_conf_dict)
  File "/pai/pai-management/paiLibrary/common/template_handler.py", line 28, in generate_from_template_dict
    map_table
  File "/usr/lib/python2.7/dist-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/lib/python2.7/dist-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "<template>", line 45, in top-level template code
  File "/usr/lib/python2.7/dist-packages/jinja2/environment.py", line 389, in getitem
    return obj[argument]
UndefinedError: 'dict object' has no attribute 'alerting'
Traceback (most recent call last):
  File "paictl.py", line 609, in <module>
    main()
  File "paictl.py", line 594, in main
    pai_service()
  File "paictl.py", line 468, in pai_service
    service_management_starter.run()
  File "/pai/pai-management/paiLibrary/paiService/service_management_start.py", line 111, in run
    self.start(serv)
  File "/pai/pai-management/paiLibrary/paiService/service_management_start.py", line 84, in start
    service_template_generater.run()
  File "/pai/pai-management/paiLibrary/paiService/service_template_generate.py", line 95, in run
    self.generate_template()
  File "/pai/pai-management/paiLibrary/paiService/service_template_generate.py", line 85, in generate_template
    raise e
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'alerting'
```

--- a/pai-management/bootstrap/prometheus/alert-deployment.yaml.template
+++ b/pai-management/bootstrap/prometheus/alert-deployment.yaml.template
@@ -15,6 +15,13 @@
 # DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
+{% set prom_info = clusterinfo['prometheusinfo'] %}
+{% if 'alerting' in prom_info and 'alert_manager_port' in prom_info['alerting'] %}
+{% set port = clusterinfo['prometheusinfo']['alerting']['alert_manager_port'] %}
+{% else %}
+{% set port = 9093 %}
+{% endif %}
+
 apiVersion: apps/v1
 kind: Deployment
 metadata:
@@ -42,7 +49,7 @@ spec:
           - '--storage.path=/alertmanager'
         ports:
         - name: alertmanager
-          containerPort: {{ clusterinfo['prometheusinfo']['alerting']['alert_manager_port']|default(9093) }}
+          containerPort: {{ port }}
         volumeMounts:
         - name: config-volume
           mountPath: /etc/alertmanager

--- a/pai-management/bootstrap/prometheus/prometheus-configmap.yaml.template
+++ b/pai-management/bootstrap/prometheus/prometheus-configmap.yaml.template
@@ -15,6 +15,13 @@
 # DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
+{% set prom_info = clusterinfo['prometheusinfo'] %}
+{% if 'alerting' in prom_info and 'alert_manager_port' in prom_info['alerting'] %}
+{% set port = clusterinfo['prometheusinfo']['alerting']['alert_manager_port'] %}
+{% else %}
+{% set port = 9093 %}
+{% endif %}
+
 apiVersion: v1
 kind: ConfigMap
 metadata:
@@ -37,12 +44,12 @@ data:
         regex: '([^:]+)(:\d*)?'
         replacement: '${1}:{{ clusterinfo['prometheusinfo']['node_exporter_port'] }}'
         target_label: __address__
-{% if clusterinfo['prometheusinfo']['alert-manager-hosts'] %}
+{% if clusterinfo['prometheusinfo']['alerting'] %}
     alerting:
       alertmanagers:
         - static_configs:
           - targets:
-            {% for host in clusterinfo['prometheusinfo']['alert-manager-hosts'] %}
-            - {{ host }}:{{ clusterinfo['prometheusinfo']['alerting']['alert_manager_port']|default(9093) }}
+            {% for host in clusterinfo['prometheusinfo']['alerting']['alert-manager-hosts'] %}
+            - {{ host }}:{{ port }}
             {% endfor %}
 {% endif %}

--- a/pai-management/bootstrap/prometheus/service.yaml
+++ b/pai-management/bootstrap/prometheus/service.yaml
@@ -24,6 +24,7 @@ template-list:
   - node-exporter-ds.yaml
   - prometheus-configmap.yaml
   - prometheus-deployment.yaml
+  - start.sh
   - stop.sh
   - refresh.sh
   - delete.yaml

--- a/pai-management/bootstrap/prometheus/start.sh.template
+++ b/pai-management/bootstrap/prometheus/start.sh.template
@@ -30,8 +30,9 @@ kubectl create configmap prometheus-alert --from-file=../../../prometheus/promet
 kubectl create -f node-exporter-ds.yaml
 kubectl create -f prometheus-deployment.yaml
 kubectl create -f watchdog-ds.yaml
-kubectl create -f watchdog-ds.yaml
+{% if clusterinfo['prometheusinfo']['alerting'] %}
 kubectl create -f alert-configmap.yaml
 kubectl create -f alert-deployment.yaml
+{% endif %}
 
 popd > /dev/null

--- a/pai-management/paiLibrary/clusterObjectModel/paiObjectModel.py
+++ b/pai-management/paiLibrary/clusterObjectModel/paiObjectModel.py
@@ -285,7 +285,10 @@ class paiObjectModel:
 
             alert_manager_hosts.append(host["hostip"])
 
-        serviceDict["clusterinfo"]["prometheusinfo"]["alert-manager-hosts"] = alert_manager_hosts
+        # template can check clusterinfo['prometheusinfo']['alerting'] to see if alert is enabled
+        if serviceDict["clusterinfo"]["prometheusinfo"].get("alerting") is not None:
+            serviceDict["clusterinfo"]["prometheusinfo"]["alerting"]["alert-manager-hosts"] = \
+                    alert_manager_hosts
 
         # section
 

